spring-integration-ip/src/main/java/org/springframework/integration/ip/tcp/TcpInboundGateway.java

```diff
@@ -128,35 +128,33 @@ private boolean doOnMessage(Message<?> message) {
 			return false;
 		}
 		String connectionId = (String) message.getHeaders().get(IpHeaders.CONNECTION_ID);
-		TcpConnection connection = null;
 		if (connectionId != null) {
-			connection = this.connections.get(connectionId);
-		}
-		if (connection == null) {
-			publishNoConnectionEvent(message, connectionId);
-			logger.error(() -> "Connection not found when processing reply " + reply + " for " + message);
-			return false;
-		}
-		try {
-			connection.send(reply);
-		}
-		catch (Exception ex) {
-			logger.error(ex, "Failed to send reply");
+			TcpConnection connection = this.connections.get(connectionId);
+			if (connection == null) {
+				publishNoConnectionEvent(message, connectionId);
+				logger.error(() -> "Connection not found when processing reply " + reply + " for " + message);
+				return false;
```

Comment:
Just curious: This method always returns false, but I didn't look into detail. If this is desired, I wondered if for the sake of readability, this one should be changed to void.

Reply:
Good catch, @mrpiggi !

The contract is this:
```
@FunctionalInterface
public interface TcpListener {

	/**
	 * Called by a TCPConnection when a new message arrives.
	 * @param message The message.
	 * @return true if the message was intercepted
	 */
	boolean onMessage(Message<?> message);

}
```
So, we need to revise and return `true` at some point.

Reply:
FYI: https://github.com/spring-projects/spring-integration/pull/10315